### PR TITLE
Fix exception caught while debugging

### PR DESCRIPTION
**Pull request by DavidKarlas**
While I was editing .cshtml this code threw exception often, so I fixed it the way I think it should be to never throw exception

### CODE REVIEW

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Navigation/TextFileNavigationPoint.cs [hunk 1]
```diff
@@ -58,10 +58,12 @@ public TextFileNavigationPoint (Document doc, ITextView textView)
...
 		{
-			if (textView != null && offset.HasValue) {
-				var currentLine = textView.TextBuffer.CurrentSnapshot.GetLineFromPosition (offset.Value);
+			if (textView != null && offset is SnapshotPoint sp) {
+				var currentSnapshot = sp.Snapshot.TextBuffer.CurrentSnapshot;
+				var position = sp.TranslateTo (currentSnapshot, PointTrackingMode.Negative).Position;
+				var currentLine = currentSnapshot.GetLineFromPosition (position);
 				line = currentLine.LineNumber;
-				column = offset.Value.Position - currentLine.Start.Position;
+				column = position - currentLine.Start.Position;
```
Comment (netonjm) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Navigation/TextFileNavigationPoint.cs]: maybe we could ensure here we didn't get here a negative value or log in that case

Comment (DavidKarlas) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Navigation/TextFileNavigationPoint.cs]: This can't be negative

Comment (netonjm) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Navigation/TextFileNavigationPoint.cs]: then 👍

**Comment by DavidKarlas** [comment]
@monojenkins backport release-8.4

**Pull request merged**